plugins/repository-s3/src/test/java/org/opensearch/repositories/s3/S3BlobContainerRetriesTests.java

```diff
@@ -280,11 +282,26 @@ public void testWriteBlobWithRetries() throws Exception {
 
         final BlobContainer blobContainer = createBlobContainer(maxRetries, null, true, null);
         try (InputStream stream = new ByteArrayInputStream(bytes)) {
-            blobContainer.writeBlob("write_blob_max_retries", stream, bytes.length, false);
+            if (metadata != null) {
+                blobContainer.writeBlobWithMetadata("write_blob_max_retries", stream, metadata, bytes.length, false);
+            } else {
+                blobContainer.writeBlob("write_blob_max_retries", stream, bytes.length, false);
+            }
         }
         assertThat(countDown.isCountedDown(), is(true));
     }
 
+    public void testWriteBlobWithMetadataWithRetries() throws Exception {
```

Comment:
What is that we are checking with this test?

Reply:
These tests were there before for method `writeBlob`. I extended this test for `writeBlob` and `writeBlobWithMetadata` to verify there is no regression problem.